service/lib/dinstaller/dbus/manager.rb

```diff
@@ -20,68 +20,79 @@
 # find current contact information at www.suse.com.
 
 require "dbus"
-require "dinstaller/manager"
 
 module DInstaller
   module DBus
-    # YaST D-Bus object (/org/opensuse/DInstaller/Manager1)
-    #
-    # @see https://rubygems.org/gems/ruby-dbus
+    # D-Bus object to manage the installation process
     class Manager < ::DBus::Object
       PATH = "/org/opensuse/DInstaller/Manager1"
       private_constant :PATH
 
       MANAGER_INTERFACE = "org.opensuse.DInstaller.Manager1"
       private_constant :MANAGER_INTERFACE
 
-      attr_reader :logger
-
+      # Constructor
+      #
+      # @param backend [DInstaller::Manager] Installation manager
       # @param logger [Logger]
-      def initialize(logger)
+      def initialize(backend, logger)
         @logger = logger
-        @backend = ::DInstaller::Manager.instance.tap(&:probe)
-        @backend.progress.add_on_change_callback do
-          PropertiesChanged(MANAGER_INTERFACE, { "Progress" => progress }, [])
-        end
-        @backend.add_status_callback do
-          PropertiesChanged(MANAGER_INTERFACE, { "Status" => @backend.status.id }, [])
-        end
+        @backend = backend
+
+        add_status_callback
+
+        add_progress_callback
 
         super(PATH)
       end
 
       dbus_interface MANAGER_INTERFACE do
-        dbus_method :Probe, "" do
-          @backend.probe
-        end
+        dbus_method(:Probe, "") { backend.probe }
 
-        dbus_method :Commit, "" do
-          @backend.install
-        end
+        dbus_method(:Commit, "") { backend.install }
 
-        # Enum list for statuses. Possible values:
-        # 0 : error ( it can be read from progress message )
-        # 1 : probing
-        # 2 : probed
-        # 3 : installing
-        # 4 : installed
+        # Current status
+        #
+        # Possible values:
```

Comment:
What about pointing to the `Status` class? I fear that these values might get outdated sooner than later.

Reply:
Yes, this is something I was thinking about. For now, I decided to keep it as it was (using the id). I am going to add a TODO note.